### PR TITLE
Make it possible for template to output imports

### LINKED ISSUE #21: Imports of modules with custom names are not included into generated code
The generator does not work with the case when the name of the package imported into source file does not match last import path segment.

For example, source file like

```go
package authz

import (
	"mymodulename/pkg/api/iam/v1" // Go package name is "iamv1"
)

type Enforcer interface {
	Enforce(ctx context.Context, request *iamv1.Request) ([]string, error)
}
```

Will result generated code that is using `*iamv1.Request` type in method signatures, however `mymodulename/pkg/api/iam/v1` will not be added to the imports section.

I tried all sorts of aliases for this module, however, nothing helps. Generator only works if I rename imported package to `v1`.

### PR DESCRIPTION
**Pull request by recht**
Sometimes, especially in large projects, goimports cannot properly resolve the imports by itself. This adds a helper function that can be used in templates, which outputs the imports from the original file as well as any additional imports required by the template itself.

I believe this also fixes #21.

### CODE REVIEW

**Comment by kop** [comment]
I've tested this PR against #21 and can confirm that it solves the issue.

File: generator/generator.go [hunk 1]
```diff
@@ -35,7 +35,28 @@ type TemplateInputs struct {
...
+	Vars    map[string]interface{}
+	Imports []string
+}
+
+// RenderImports outputs a list of imports, combining the ones from
+// the source file with any provided for the template itself.
+func (t TemplateInputs) RenderImports(imports ...string) string {
+	allImports := make(map[string]struct{})
+	for _, i := range t.Imports {
+		allImports[i] = struct{}{}
```
Comment (kop) [inline generator/generator.go:47]:
```suggestion
		allImports[strings.TrimSpace(i)] = struct{}{}
```

File: generator/generator.go [hunk 2]
```diff
@@ -35,7 +35,28 @@ type TemplateInputs struct {
...
+	}
+	for _, i := range imports {
+		if i[len(i)-1] != '"' {
+			i = `"` + i + `"`
+		}
+		allImports[i] = struct{}{}
```
Comment (kop) [inline generator/generator.go:53]:
```suggestion
		allImports[strings.TrimSpace(i)] = struct{}{}
```

**Comment by hexdigest** [comment]
Hi @recht,

Thank you for this wonderful PR! The only thing I want to ask is to provide a usage example by changing one of the templates within the templates dir.

@kop thank you for testing this!

File: generator/generator.go [hunk 3]
```diff
@@ -35,7 +35,28 @@ type TemplateInputs struct {
...
+func (t TemplateInputs) RenderImports(imports ...string) string {
+	allImports := make(map[string]struct{})
+	for _, i := range t.Imports {
+		allImports[strings.TrimSpace(i)] = struct{}{}
+	}
+	for _, i := range imports {
+		if i[len(i)-1] != '"' {
+			i = `"` + i + `"`
+		}
+		allImports[strings.TrimSpace(i)] = struct{}{}
```
Comment (hexdigest) [inline generator/generator.go:53]: @recht shouldn't we trim space from the user input first and then add quotes?
If you can fix this I'm ok to fix templates myself using this new helper.

**Review by hexdigest — CHANGES_REQUESTED** [review]

**Comment by hexdigest** [comment]
@recht jfyi I merged this PR to render_imports branch.
I'll fix some issues there and then it will be merged to master.

**Comment by hexdigest** [comment]
@recht @kop 

Please check v1.1.10 the name of the helper is "Import"

**Pull request merged**